sentinel5dl/__main__.py

```diff
@@ -7,30 +7,149 @@
 :url: https://emissions-api.org
 :license: MIT
 '''
-
+import argparse
+import iso8601
 import logging
+import textwrap
 
 import sentinel5dl
 from sentinel5dl import search, download
 
+PRODUCTS = (
+    'L1B_IR_SIR',
+    'L1B_IR_UVN',
+    'L1B_RA_BD1',
+    'L1B_RA_BD2',
+    'L1B_RA_BD3',
+    'L1B_RA_BD4',
+    'L1B_RA_BD5',
+    'L1B_RA_BD6',
+    'L1B_RA_BD7',
+    'L1B_RA_BD8',
+    'L2__AER_AI',
+    'L2__AER_LH',
+    'L2__CH4___',
+    'L2__CLOUD_',
+    'L2__CO___',
+    'L2__HCHO__',
+    'L2__NO2__',
+    'L2__NP_BD3',
+    'L2__NP_BD6',
+    'L2__NP_BD7',
+    'L2__O3_TCL',
+    'L2__O3___',
+    'L2__SO2__',
+)
+
+PRODUCTS_STR = textwrap.fill(', '.join(PRODUCTS),
+                             subsequent_indent='  ',
+                             initial_indent='  ')
+
+PROCESSING_LEVELS = (
+    'L1B',
+    'L2'
+)
+
+
+def is_polygon(polygon):
```

Comment:
I've basically done two things to this method:

1. Catch some more error cases like the values being non-floats
2. Instead of trying to build complex lists of booleans and checking them at the end yet again, I'm raising a `ValueError` as soon as we detect an error. That way we bail straight out of the method and spare us a lot of unnecessary code.